springboot/src/main/java/ca/jrvs/apps/trading/dao/QuoteDao.java

```diff
@@ -1,7 +1,222 @@
+   */
+  private int updateOne(Quote quote) {
+    String update_sql = "UPDATE quote SET last_price=?, bid_price=?, "
+        + "bid_size=?, ask_price=?, ask_size=? WHERE ticker=?";
+    return jdbcTemplate.update(update_sql, makeUpdateValues(quote));
+  }
+
+  /**
+   * Helper metehod that makes sql update values objects
+   *
+   * @param quote to be updated
+   * @return UPDATE_SQL values
+   */
+  private Object[] makeUpdateValues(Quote quote) {
+    int num = quote.getClass().getDeclaredFields().length;
+    Object[] obj = new Object[num];
+    obj[0] = quote.getLastPrice();
+    obj[1] = quote.getBidPrice();
+    obj[2] = quote.getBidSize();
+    obj[3] = quote.getAskPrice();
+    obj[4] = quote.getAskSize();
+    obj[5] = quote.getTicker();
+    return obj;
+  }
+
+  /**
+   * Helper function to run all queries
+   *
+   * @param allQuotes list of Quotes to update/insert
+   * @return outputQuotes list of Quotes run in table
+   */
+  @Override
+  public <S extends Quote> Iterable<S> saveAll(Iterable<S> allQuotes) {
+    List<Quote> outputQuotes = new ArrayList<Quote>();
+    //run save for all quotes in list and append to array
+    for (Quote q : allQuotes) {
+      outputQuotes.add(save(q));
+    }
+    return (Iterable<S>) outputQuotes;
+  }
+
+  /**
+   * Find a quote by ticker
+   *
+   * @param ticker name
+   * @return quote or Option.empty if not found
+   */
+  @Override
+  public Optional<Quote> findById(String ticker) {
+    Optional<Quote> findID = Optional.empty();
+    String selectSql = "SELECT * FROM " + TABLE_NAME + " WHERE " + ID_COLUMN_NAME + "=?";
+    //Advanced: handle read + update race condition
+    Quote quote = null;
+    try {
+      quote = jdbcTemplate.queryForObject(
+          selectSql,
+          new Object[]{ticker},
+          new RowMapper<Quote>() {
+            public Quote mapRow(ResultSet rs, int rowNum) throws SQLException {
+              Quote innerQuote = new Quote();
+              innerQuote.setTicker(rs.getString("ticker"));
+              innerQuote.setLastPrice(rs.getDouble("last_price"));
+              innerQuote.setBidPrice(rs.getDouble("bid_price"));
+              innerQuote.setBidSize(rs.getLong("bid_size"));
+              innerQuote.setAskPrice(rs.getDouble("ask_price"));
+              innerQuote.setAskSize(rs.getLong("ask_size"));
+              return innerQuote;
+            }
+          });
+    } catch (Exception e) {
+      logger.debug("Can't find ticker: " + ticker, e);
+    }
+    return Optional.of(quote);
+  }
+
+  /**
+   * Check if ticker exists
+   *
+   * @param ticker id String
+   * @return boolean True if exists, False if not
+   */
+  @Override
+  public boolean existsById(String ticker) {
+    String selectSql = "SELECT COUNT(*) FROM " + TABLE_NAME + " WHERE " + ID_COLUMN_NAME + "=?";
+    boolean exists = false;
+    long count = jdbcTemplate.queryForObject(selectSql, new Object[]{ticker}, Long.class);
+    return (count > 0) ? true : false;
```

Comment:
`exists` appears unused. You can probably save some work here querying for a RowSet and returning RowSet.next().
The cursor for RowSet starts BEFORE the first row, so RowSet.next() will return true if there's at least one row.

Alternatively you can just `return count > 0` since the comparison creates a boolean